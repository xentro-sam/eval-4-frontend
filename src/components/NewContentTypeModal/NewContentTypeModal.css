#new-content-type-modal {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    font: 1vw;
}

#modal-content {
    background-color: white;
    width: 35vw;
    height: 15vw;
    border-radius: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
    padding: 0.25vw 3vw;
}

#modal-title {
    font-size: 1.5vw;
    font-weight: 600;
    margin-bottom: 1vw;
}

#modal-subtitle {
    font-size: 1vw;
    font-weight: 400;
    margin-bottom: 0.25vw;
}

#modal-input > input {
    width: 100%;
    height: 2vw;
    border: 1px solid rgb(200, 210, 250);
    border-radius: 4px;
    box-sizing: border-box;
    font-size: 1vw;
    padding: 0.25vw;

}

#modal-buttons {
    display: flex;
    justify-content: flex-end;
    margin-top: 1vw;
}

#modal-cancel {
    width: 5vw;
    padding-top: 0.5vw;
}

#modal-create {
    width: 10vw;
    height: 2.5vw;
    margin-left: 1vw;
    border-radius: 5px;
    background-color: rgb(122, 97, 246);
    color: white;
}