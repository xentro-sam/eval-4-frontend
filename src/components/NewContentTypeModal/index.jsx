import * as React from 'react';
import './NewContentTypeModal.css';
import PropTypes from 'prop-types';
import makeRequest from '../../utils/makeRequest';
import {CREATE_CONTENT_TYPE, GET_CONTENT_TYPES} from '../../constants/apiEndPoints';

export default function NewContentTypeModal(props) {
  if (!props.show) {
    return null;
  }
  const handleClick = async () => {
    const newFieldName = document.getElementById('modal-input').children[0].value;
    await makeRequest(CREATE_CONTENT_TYPE, {
      data: {
        contentTypeName: newFieldName,
        contentTypeFields: [],
      },
    });
    const updatedContentTypes = await makeRequest(GET_CONTENT_TYPES, {});
    await props.setContentTypes(updatedContentTypes);
    await props.setCollectionTypes(updatedContentTypes);
  };
  return (
    <div id="new-content-type-modal">
      <div id="modal-content">
        <div id="modal-title">
            Create a new content type
        </div>
        <div id="modal-subtitle">
            Name of the content type
        </div>
        <div id="modal-input">
          <input type="text" />
        </div>
        <div id="modal-buttons">
          <div id="modal-cancel" onClick={props.onClose}>Cancel</div>
          <button type="submit" id="modal-create" onClick={handleClick}>Create</button>
        </div>
      </div>
    </div>
  );
}

NewContentTypeModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool,
  setContentTypes: PropTypes.func.isRequired,
  setCollectionTypes: PropTypes.func.isRequired,
};
