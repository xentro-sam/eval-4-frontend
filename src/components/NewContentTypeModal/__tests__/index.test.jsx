import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NewContentTypeModal from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  onClose: jest.fn(),
  show: true,
  setContentTypes: jest.fn(),
  setCollectionTypes: jest.fn(),
};

describe('ContentType', () => {
  it('should render correctly', async () => {
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<NewContentTypeModal {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('Name of the content type')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
