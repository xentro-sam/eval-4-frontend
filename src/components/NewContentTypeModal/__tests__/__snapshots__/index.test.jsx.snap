// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="new-content-type-modal"
  >
    <div
      id="modal-content"
    >
      <div
        id="modal-title"
      >
        Create a new content type
      </div>
      <div
        id="modal-subtitle"
      >
        Name of the content type
      </div>
      <div
        id="modal-input"
      >
        <input
          type="text"
        />
      </div>
      <div
        id="modal-buttons"
      >
        <div
          id="modal-cancel"
        >
          Cancel
        </div>
        <button
          id="modal-create"
          type="submit"
        >
          Create
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
